src/coreclr/tools/Common/Internal/Metadata/NativeFormat/Generator/ReaderGen.cs

```diff
@@ -100,7 +100,7 @@ private void EmitHandle(RecordDef record)
     {
         string handleName = $"{record.Name}Handle";
 
-        OpenScope($"public partial struct {handleName}");
+        OpenScope($"public partial struct {handleName} : IEquatable<{handleName}>");
```

Comment:
It may be better to omit `IEquatable` for NativeFormat reader. NativeFormat is internal implementation detail, it is a lot of structs and I am not sure whether the linker/compiler is going to be able to trim it.

Reply:
Ok, I can revert these and suppress the warnings if that's the preference.